src/greptimeai/tracker/openai_tracker.py

```diff
@@ -103,6 +117,38 @@ def _patch(self, extractor: BaseExtractor):
         if not func:
             return
 
+        if extractor.is_async():
+            @functools.wraps(func)
+            async def wrapper(*args, **kwargs):
+                req_extraction = extractor.pre_extract(*args, **kwargs)
+                span_id = self.start_span(extractor.get_span_name(), req_extraction)
+                common_attrs = {_SPAN_NAME_LABEL: extractor.get_span_name()}
+                start = time.time()
+                ex = None
+                try:
+                    resp = await func(*args, **kwargs)
```

Comment:
can you try multiple async methods calling? I'm afraid this may transfer users' async methods into sync ones by accident.

Reply:
and if this is the only difference with sync version, can we change this to:

```
if extractor.is_async():
    resp = await func(*args, **kwargs)
else:
    resp = func(*args, **kwargs)
```

Reply:
we cannot use `await` keyword in common sync def